datafusion/core/src/catalog/mod.rs

```diff
@@ -29,7 +29,10 @@ use datafusion_common::{exec_err, not_impl_err, DataFusionError, Result};
 use std::any::Any;
 use std::sync::Arc;
 
-/// Represent a list of named catalogs
+/// Represent a list of named [`CatalogProvider`]s.
+///
+/// Please see the documentation on `CatalogProvider` for details of
+/// implementing a custom catalog.
 pub trait CatalogList: Sync + Send {
```

Comment:
perhaps CatalogProviderList will be a better name? I can do this in following PR

Reply:
Filed https://github.com/apache/arrow-datafusion/pull/9002